app/templating/summary/question.py

```diff
@@ -24,13 +24,20 @@ def _build_answers(cls, block_id, question_schema, answer_schema, answers):
         summary_answers = []
         answers_iterator = iter(answer_schema)
         for answer_schema in answers_iterator:
-            answer = cls._build_answer(question_schema, answer_schema, answers, answers_iterator)
-            summary_answers.append(Answer(block_id, answer_schema, answer))
+            if question_schema['type'] == 'RepeatingAnswer':
+                for k, _ in answers.items():
+                    if k.startswith(answer_schema['id']):
```

Comment:
this simple check will suffer the same problems as the original code before ```_extract_answer_instance_id``` was introduced; could re-use ```_iterate_over_instance_ids```; if it is modified to take user_input.keys() instead of user_input then it could also take answers.keys() here as well to get the same effect?